deepchem/models/torch_models/smiles2vec.py

```diff
@@ -0,0 +1,176 @@
+import torch.nn as nn
+import math
+from typing import List, Dict, Optional
+
+
+class Smiles2Vec(nn.Module):
+    """
+    Implements the Smiles2Vec model, that learns neural representations of SMILES
+    strings which can be used for downstream tasks.
+
+    The model is based on the description in Goh et al., "SMILES2vec: An
+    Interpretable General-Purpose Deep Neural Network for Predicting Chemical
+    Properties" (https://arxiv.org/pdf/1712.02034.pdf). The goal here is to take
```

Comment:
@harishwar017 We always keep the references in the end of the doc strings.

References
----------------
..[1] SMILES2vec: An
    Interpretable General-Purpose Deep Neural Network for Predicting Chemical
    Properties" (https://arxiv.org/pdf/1712.02034.pdf)
    
In this way, pls add these changes to it.